src/Entity/GeneratedBook.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace App\Entity;
+
+use App\Book;
+use DateTime;
+use DateTimeZone;
+// mediawiki-codesniffer apparently can't search annotations for unused classes.
+// phpcs:ignore MediaWiki.Classes.UnusedUseStatement.UnusedUse
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * The GeneratedBook is the log of exports.
+ * @ORM\Entity(repositoryClass="App\Repository\GeneratedBookRepository")
+ * @ORM\Table(
+ *     name="books_generated",
+ *     indexes={
+ * @ORM\Index(name="time", columns={"time"}),
+ * @ORM\Index(name="lang", columns={"lang"})
+ *     }
+ * )
+ */
+class GeneratedBook {
+
+	/**
+	 * @ORM\Id()
+	 * @ORM\GeneratedValue()
+	 * @ORM\Column(type="integer")
+	 * @var int
+	 */
+	private $id;
+
+	/**
+	 * @ORM\Column(type="datetime")
+	 * @var DateTime Time of the export.
+	 */
+	private $time;
+
+	/**
+	 * @ORM\Column(type="string", length=10)
+	 * @var string Language code.
+	 */
+	private $lang;
+
+	/**
+	 * @ORM\Column(type="string", length=255)
+	 * @var string Title of work.
+	 */
+	private $title;
+
+	/**
+	 * @ORM\Column(type="string", length=10)
+	 * @var string Format such as 'epub', 'pdf', etc.
+	 */
+	private $format;
+
+	/**
+	 * GeneratedBook constructor.
+	 * @param Book $book
+	 * @param string $format
+	 */
+	public function __construct( Book $book, string $format ) {
+		$this->time = new DateTime( 'now', new DateTimeZone( 'UTC' ) );
```

Comment:
Why not use setters for all of these? It reads a bit oddly that only Title does so.

Reply:
MUSIKANIMAL IS A WIKIPEDOPHILE